packages/components/pie-list/src/pie-list-item/index.ts

```diff
@@ -1,15 +1,76 @@
-import { LitElement } from 'lit';
-import { safeCustomElement } from '@justeattakeaway/pie-webc-core';
-import { type ListItemProps } from './defs';
+import { html, unsafeCSS } from 'lit';
+import { PieElement } from '@justeattakeaway/pie-webc-core/src/internals/PieElement';
+import { RtlMixin, safeCustomElement } from '@justeattakeaway/pie-webc-core';
+import { property } from 'lit/decorators.js';
+import type { PropertyValues } from 'lit';
+
+import styles from './list-item.scss?inline';
+import { type ListItemProps, defaultProps } from './defs';
+
+// Valid values available to consumers
+export * from './defs';
 
 const componentSelector = 'pie-list-item';
 
 /**
  * @tagname pie-list-item
+ *
+ * PIE List Item component for use within pie-list.
  */
 @safeCustomElement('pie-list-item')
-export class PieListItem extends LitElement implements ListItemProps {
-    // component logic
+export class PieListItem extends RtlMixin(PieElement) implements ListItemProps {
+    @property({ type: String })
+    public primaryText = defaultProps.primaryText;
+
+    @property({ type: Boolean, reflect: true })
+    public selected = defaultProps.selected;
+
+    @property({ type: Boolean, reflect: true })
+    public disabled = defaultProps.disabled;
+
+    connectedCallback () {
+        super.connectedCallback();
+
+        if (!this.hasAttribute('role')) {
+            this.setAttribute('role', 'listitem');
+        }
+
+        if (!this.hasAttribute('data-test-id')) {
+            this.setAttribute('data-test-id', 'pie-list-item');
```

Comment:
I understand that this only happens if none is provided by consumers, but it would lead to duplicate test-ids. Unsure if we should add a `uuid` or something (or not self-render a test ID at all?